File: src/main/java/com/example/controller/Responses/FestiveInfo.java
package com.example.controller.Responses;

import java.util.ArrayList;
import java.util.Calendar;
import java.util.List;

public class FestiveInfo {
    private int day;
    private int month;
    private int year;
    private String color;
    private int startedXDaysAgo;
    private String local;

    public FestiveInfo(int day, int month, int year, String color, int startedXDaysAgo, String local) {
        this.day = day;
        this.month = month;
        this.year = year;
        this.color = color;
        this.startedXDaysAgo = startedXDaysAgo;
        this.local = local;
    }

    public static List<FestiveInfo> generateFestivesFromRange(FestiveInfo festive) {
        List<FestiveInfo> festiveList = new ArrayList<>();

        // Crear la fecha final (el día del festivo especificado)
        Calendar endDate = Calendar.getInstance();
        endDate.set(festive.getYear(), festive.getMonth() - 1, festive.getDay());
        
        // Calcular la fecha de inicio restando los xdaysAgo
        Calendar startDate = (Calendar) endDate.clone();
        startDate.add(Calendar.DAY_OF_MONTH, -festive.getStartedXDaysAgo());
        
        // Iterar desde la fecha de inicio hasta la fecha final
        Calendar currentDate = (Calendar) startDate.clone();
        while (!currentDate.after(endDate)) {
            // Crear un nuevo objeto FestiveInfo para cada día
            festiveList.add(new FestiveInfo(
                currentDate.get(Calendar.DAY_OF_MONTH),
                currentDate.get(Calendar.MONTH) + 1, // Ajustar el mes al rango 1-12
                currentDate.get(Calendar.YEAR),
                festive.getColor(), // Usar el mismo color
                festive.getStartedXDaysAgo(),
                festive.getLocal() // Usar la misma localidad
            ));
            
            // Avanzar un día
            currentDate.add(Calendar.DAY_OF_MONTH, 1);
        }
        return festiveList;
    }

    public int getDay() {
        return day;
    }

    public int getMonth() {
        return month;
    }

    public int getYear() {
        return year;
    }

    public String getColor() {
        return color;
    }

    public int getStartedXDaysAgo() {
        return startedXDaysAgo;
    }

    public String getLocal() {
        return local;
    }
}

File: MyURJC/src/app/services/FestiveService/FestiveInfo.ts
export class FestiveInfo {
    day: number;
    month: number;
    year: number;
    color: string;
    startedXDaysAgo: number;
    local: string

    constructor(day: number, month: number, year: number, color: string, startedXDaysAgo: number, local: string) {
        this.day = day;
        this.month = month;
        this.year = year;
        this.color = color;
        this.startedXDaysAgo = startedXDaysAgo;
        this.local = local;



    }

    static generateFestivesFromRange(festive: FestiveInfo): FestiveInfo[] {
        const festiveList: FestiveInfo[] = [];

        // Crear la fecha final (el día del festivo especificado)
        const endDate = new Date(festive.year, festive.month - 1, festive.day);

        // Calcular la fecha de inicio restando los xdaysAgo
        const startDate = new Date(endDate);
        startDate.setDate(endDate.getDate() - festive.startedXDaysAgo);

        // Iterar desde la fecha de inicio hasta la fecha final
        let currentDate = new Date(startDate);
        while (currentDate <= endDate) {
            // Crear un nuevo objeto FestiveInfo para cada día
            festiveList.push(new FestiveInfo(
                currentDate.getDate(),
                currentDate.getMonth() + 1, // Ajustar el mes al rango 1-12
                currentDate.getFullYear(),
                festive.color, // Usar el mismo color
                festive.startedXDaysAgo,
                festive.local // Usar la misma localidad
            ));

            // Avanzar un día
            currentDate.setDate(currentDate.getDate() + 1);
        }
        return festiveList;
    }


}

